### ISSUE TITLE
Cannot build image from institutional Figshare DOIs

### ISSUE THREAD

**Issue by zoidy** [jupyterhub/repo2docker#947]
### Bug description
While building images from DOIs from the public figshare.com repository, things work great (e.g., https://doi.org/10.6084/m9.figshare.12746861.v1). However, entering a DOI from an institutional Figshare instance (e.g., https://doi.org/10.25422/azu.data.12456536.v1) does not work. The error given is
`Error during build: Failed to clone repository from 10.25422/azu.data.12456536.v1 (ref 12456536.v1)`

Institutional Figshare instances may have different URLs but all Figshare API calls are to api.figshare.com/v2/ even for institutional instances, so I my assumption was that it would work.

Related: jupyterhub/binderhub#938 jupyterhub/binderhub#951 

#### Expected behaviour
A Docker image is successfully created

#### Actual behaviour
The Docker image does not build.

### How to reproduce
Enter the above test DOIs on MyBinder.org to see the behavior

**Comment by nuest** [comment]
@betatim I agree with your analysis of the problem, and the suggested solution :-)
